examples/Chainweb/Simulate/Contracts/CoinContract.hs

```diff
@@ -0,0 +1,196 @@
+
+instance Show CoinContractRequest where
+  show (CoinCreateAccount account _guard) = "CoinContract.CoinCreateAccount: " ++ parens (show account)
+  show (CoinAccountBalance account) = "CoinContract.CoinAccountBalance: " ++ parens (show account)
+  show (CoinTransfer sender receiver _guard amount) =
+    "CoinContract.CoinTransfer: "
+    ++ parens (show sender)
+    ++ parens (show receiver)
+    ++ parens (show amount)
+
+type Guard = [SomeKeyPair]
+
+-- for simplicity
+type SenderName = Account
+type ReceiverName = Account
+type MinerName = Account
+type Name = Account
+newtype Address = Address String
+
+errPrefix :: String
+errPrefix = "CoinContract."
+
+mkRandomCoinContractRequest :: [(Account,Maybe [SomeKeyPair])] -> IO (FGen CoinContractRequest)
+mkRandomCoinContractRequest kacts = do
+  request <- randomRIO (0, 1 :: Int)
+  return $
+    case request of
+      0 -> CoinAccountBalance <$> fake
+      1 -> do
+        (from, to) <- distinctPair
+        case join (lookup to kacts) of
+          Nothing -> error (errmsg ++ getAccount to)
+          Just keyset -> CoinTransfer from to keyset <$> fake
+      _ -> error "mkRandomCoinContractRequest: impossible case"
+  where
+    errmsg =
+      "mkRandomCoinContractRequest: something went wrong." ++
+      " Cannot find account name"
+
+createCoinContractRequest :: PublicMeta -> CoinContractRequest -> IO (Command Text)
+createCoinContractRequest meta request =
+  case request of
+    CoinCreateAccount (Account account) guard -> do
+      adminKeyset <- testSomeKeyPairs
+      let theCode =
+            printf
+            "(coin.create-account \"%s\" (read-keyset \"%s\"))"
+            account
+            ("create-account-guard" :: String)
+          theData =
+            object
+              [ "admin-keyset" .= fmap formatB16PubKey adminKeyset
+              , "create-account-guard" .= fmap formatB16PubKey guard
+              ]
+      mkExec theCode theData meta adminKeyset Nothing
+    CoinAccountBalance (Account account) -> do
+      adminKeyset <- testSomeKeyPairs
+      let theData = Null
+          theCode =
+            printf
+            "(coin.account-balance \"%s\")"
+            account
+      mkExec theCode theData meta adminKeyset Nothing
+    CoinTransfer (Account sendername) (Account receivername) guard (Amount amount) -> do
+      adminKeyset <- testSomeKeyPairs
+      let theCode =
+            printf
+            "(coin.transfer \"%s\" \"%s\" (read-keyset \"%s\") %s)"
+            sendername
+            receivername
+            ("receiver-guard" :: String)
+            (show amount)
+          theData =
+            object
+              [ "admin-keyset" .= fmap formatB16PubKey adminKeyset
+              , "receiver-guard" .= fmap formatB16PubKey guard
+              ]
+      mkExec theCode theData meta adminKeyset Nothing
+  where
+    _functionErrPrefix :: String
+    _functionErrPrefix = "createCoinContractRequest:"
+
+-----------------------------------
```

Comment:
You will never be issuing buy gas or coinbase, so you can delete the commented code here.

Reply:
This has been taken care of!